polaris-react/src/components/IndexTable/components/Row/Row.tsx

```diff
@@ -115,7 +115,8 @@ export const Row = memo(function Row({
 
   const rowClassName = classNames(
     styles.TableRow,
-    rowType === 'subheader' && styles['TableRow-subheader'],
+    (rowType === 'subheader' && styles['TableRow-subheader']) ||
+      (rowType === 'child' && styles['TableRow-child']),
```

Comment:
We shouldn't need this as the modifier classes exist on their own depending on a `rowType`

```suggestion
    rowType === 'subheader' && styles['TableRow-subheader'])
    rowType === 'child' && styles['TableRow-child'])
```